plottr/data/qcodes_dataset.py

```diff
@@ -139,56 +143,55 @@ def get_runs_from_db_as_dataframe(path, *arg, **kw):
     return df
 
 
-# Getting data from a dataset
+# Extracting data
 
-def get_data_from_ds(ds: DataSet, start: Optional[int] = None,
-                     end: Optional[int] = None) -> Dict[str, List[List]]:
+def ds_to_datadicts(ds: DataSet) -> Dict[str, DataDict]:
     """
-    Returns a dictionary in the format {'name' : data}, where data
-    is what dataset.get_data('name') returns, i.e., a list of lists, where
-    the inner list is the row as inserted into the DB.
+    Make DataDicts from a qcodes DataSet.
 
-    with `start` and `end` only a subset of rows in the DB can be specified.
+    :param ds: qcodes dataset
+    :returns: dictionary with one item per dependent.
+              key: name of the dependent
+              value: DataDict containing that dependent and its
+                     axes.
     """
-    names = [n for n, v in ds.paramspecs.items()]
-    return {n : np.squeeze(ds.get_data(n, start=start, end=end)) for n in names}
+    ret = {}
+    pdata = ds.get_parameter_data()
+    for p, spec in ds.paramspecs.items():
+        if spec.depends_on != '':
+            axes = spec.depends_on.split(', ')
```

Comment:
if we say that plottr works only with qcodes version 0.2.0+ (released ~7months ago) then this string-manipulation can be substituted with `spec.depends_on_` (with underscore) which returns a list of parameter names (as opposed to a comma-separated list of parameter names).

Reply:
ah, thanks! i didn't know that function. will change that!

Reply:
(i always wondered why the string format)